### PR TITLE
rlp-derive extracted

### PR DESCRIPTION
**Pull request by ordian**
`rlp_derive` can be reused by both `open-ethereum` and `substrate eth-compatibility layer`.

### CODE REVIEW

File: rlp-derive/src/de.rs [hunk 1]
```diff
@@ -0,0 +1,193 @@
+// Copyright 2015-2020 Parity Technologies (UK) Ltd.
+// This file is part of Parity Ethereum.
```
Comment (ordian) [inline rlp-derive/src/de.rs]: `s/Parity Ethereum/Parity Common` or `s/Parity Ethereum/Parity` as in other files?

Comment (dvdplm) [reply rlp-derive/src/de.rs]:
```
// Copyright 2020 Parity Technologies
//
// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
// option. This file may not be copied, modified, or distributed
// except according to those terms.
```

**Comment by niklasad1** [comment]
To bad that the history of this crate is faulty (I guess after moving the files internally in parity), 

@debris is the main contributor of this crate IIRC.

**Comment by dvdplm** [comment]
> (I guess after moving the files internally in parity),

Probably yes. `git mv` ftw.

File: rlp-derive/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+[package]
+name = "rlp_derive"
```
Comment (ordian) [inline rlp-derive/Cargo.toml]: Do we want to change the name to `rlp-derive`? (now or never)

Comment (sorpaas) [reply rlp-derive/Cargo.toml]: Please!

File: rlp-derive/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+version = "0.1.0"
+authors = ["Parity Technologies <admin@parity.io>"]
+license = "MIT/Apache-2.0"
+description = "Derive macro for #[derive(RlpEncodable, RlpDecodable)]"
+homepage = "http://parity.io"
+edition = "2018"
+
+[lib]
+name = "rlp_derive"
```
Comment (sorpaas) [inline rlp-derive/Cargo.toml]: This line is unnecessary.

File: rlp-derive/Cargo.toml (hunk 2 above)
Comment (dvdplm) [reply rlp-derive/Cargo.toml]: YESS!

**Pull request merged**